src/transformStateWithClones.js

```diff
@@ -8,6 +8,37 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const history = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = Object.assign({}, currentState, action.extraData);
+        break;
+      case 'removeProperties':
+        currentState = removeProp(currentState, action);
```

Comment:
The function 'removeProp' is called with the entire action object, but only 'keysToRemove' is necessary. Consider passing only 'keysToRemove' to the 'removeProp' function.